### PR TITLE
feat: generate SDKs for Looker 24.0

### PR DESCRIPTION
**Pull request by drstrangelooker**
I disabled the pre-commit hook in order to get this to build. Please look at diffs closely to see if it looks good.

Release-As: 24.0.0

### CODE REVIEW

**Comment by drstrangelooker** [comment]
My output @jeremytchang @jkaster 
```
$ bin/sdk_gen 24.0                                                                                                                 
/usr/local/google/home/drstrangelove/sdk-codegen/bin/sdk_gen <version>                                                                                                              
  version: Looker release version, like 22.10                                                                                                                                       
Generating SDKs for Looker 24.0 ...                                                                                                                                                 
fatal: 'refs/heads/sdk_24_0' - not a valid ref                                                                                                                                      
Creating branch sdk_24_0 ...                                                                                                                                                        
Running git checkout -b sdk_24_0 origin/main ...                                                                                                                                    
Switched to a new branch 'sdk_24_0'                                                                                                                                                 
Running yarn gen ...                                                                                                                                                                
Error: the command black could not be found within PATH or Pipfile's [scripts].                                                                                                     
Running bin/smoke typescript python kotlin ...                                                                                                                                      
PASS packages/sdk-rtl/src/delimArray.spec.ts                                                                                                                                        
PASS packages/sdk-rtl/src/browserTransport.spec.ts                                                                                                                                  
PASS packages/sdk-rtl/src/browserSession.spec.ts                                                                                                                                    
PASS packages/sdk-rtl/src/transport.spec.ts                                                                                                                                         
PASS packages/sdk-rtl/src/extensionSession.spec.ts                                                                                                                                  
PASS packages/sdk-rtl/src/paging.spec.ts
PASS packages/sdk-rtl/src/errorDoc.spec.ts                                                                                                                                 [57/9447]
PASS packages/sdk-rtl/src/constants.spec.ts                                                                                                                                         
PASS packages/sdk-rtl/src/authToken.spec.ts                                                                                                                                         
PASS packages/sdk-rtl/src/apiMethods.spec.ts                                                                                                                                        
PASS packages/sdk-rtl/src/extensionTransport.spec.ts                                                                                                                                
PASS packages/sdk-rtl/src/browserServices.spec.ts                                                                                                                                   
PASS packages/sdk-rtl/src/proxySession.spec.ts                                                                                                                                      
PASS packages/sdk-rtl/src/apiSettings.spec.ts                                                                                                                                       
PASS packages/sdk-rtl/src/oauthSession.spec.ts                                                                                                                                      
                                                                                                                                                                                    
Test Suites: 15 passed, 15 total                                                                                                                                                    
Tests:       104 passed, 104 total                                                                                                                                                  
Snapshots:   0 total                                                                                                                                                                
Time:        5.597 s                                                                                                                                                                
Ran all test suites matching /packages\/sdk-rtl/i.                                                                                                                                  
Loading .env environment variables...                                                                                                                                               
Error: the command tox could not be found within PATH or Pipfile's [scripts].                                                                                                       
                                                                                                                                                                                    
38 tests completed, 2 failed                                                                                                                                                        
                                                                                                                                                                                    
FAILURE: Build failed with an exception.                                                                                                                                            
                                                                                                                                                                                    
* What went wrong:                                                                                                                                                                  
Execution failed for task ':test'.                                                                                                                                                  
> There were failing tests. See the report at: file:///usr/local/google/home/drstrangelove/sdk-codegen/kotlin/build/reports/tests/test/index.html                                   
                                                                                                                                                                                    
* Try:                                                                                                                                                                              
> Run with --scan to get full insights.                                                                                                                                             
                                                                                                                                                                                    
BUILD FAILED in 12s                                                                                                                                                                 
PASS packages/sdk-node/test/methods.spec.ts (21.163 s)                                                                                                                              
  LookerNodeSDK                                                                                                                                                                     
    ✓ assigns SDK.ApiVersion (6 ms)                                                                                                                                                 
    issue-related tests                                                                                                                                                             
      ✓ create_user_attribute options (197 ms)                                                                                                                                      
    downloads                                                                                                                                                                       
      ✓ png and svg (207 ms)                                                                                                                                                        
    PUT smoke test                                                                                                                                                                  
      ✓ set default color collection (316 ms)                                                                                                                                       
    automatic authentication for API calls                                                                                                                                          
      ✓ me returns the correct result (131 ms)                                                                                                                                      
      ✓ me fields filter (142 ms)                                                                                                                                                   
    sudo                                                                                                                                                                            
      ✓ login/logout (1356 ms)                                                                                                                                                      
    retrieves collections                                                                                                                                                           
      ✓ search_looks returns looks (224 ms)                                                                                                                                         
      ✓ search_looks fields filter (164 ms)                                                                                                                                         
      ✓ search_looks fields and title (230 ms)                                                                                                                                      
    User CRUD-it checks                                                                                                                                                             
      ✓ create, update, and delete user (4344 ms)                                                                                                                                   
    User searches                                                                                                                                                                   
      ✓ bad search returns no results (134 ms)                                                                                                                                      
      ✓ matches email domain (195 ms)                                                                                                                                               
      ✓ csv user id list aka DelimArray (316 ms)                                                                                                                                    
      ✓ matches email domain and returns sorted (355 ms)                                                                                                                            
    Datagroups                                                                                                                                                                      
      ✓ gets all datagroups (116 ms)
    Types with enums                                                                                                                                                        [0/9447]
      ✓ CreateQueryTask serializes and deserializes (1 ms)
    paging alpha
      pager
        ○ skipped getRel can override limit and offset
        ○ skipped observers can be chained
      pageAll
        ○ skipped search_dashboard
        ○ skipped all_dashboards pageAll returns non-paged results
    Query calls
      ✓ create and run query (864 ms)
      ✓ run_inline_query (943 ms)
      ✓ parses a query with no results (2802 ms)
    Dashboard endpoints
      ✓ search_dashboards (340 ms)
      ✓ create and update dashboard (1015 ms)
    Theme
      ✓ validate_theme returns ok on valid template (117 ms)
      ✓ validate_theme throws error with details (65 ms)
    Node environment
      ✓ no INI (156 ms)

Test Suites: 1 passed, 1 total
Tests:       4 skipped, 25 passed, 29 total
Snapshots:   0 total
Time:        21.665 s, estimated 22 s
Ran all test suites matching /packages\/sdk-node\/test/i.
Loading .env environment variables...
Error: the command tox could not be found within PATH or Pipfile's [scripts].

8 tests completed, 8 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///usr/local/google/home/drstrangelove/sdk-codegen/kotlin/build/reports/tests/test/index.html

* Try:
> Run with --scan to get full insights.

BUILD FAILED in 3s
Running git add -u ...
Running git commit -m "feat: generate SDKs for Looker 24.0" -m "Release-As: 24.0.0" ...
grep: warning: stray \ before -
grep: warning: stray \ before -
Running git push origin sdk_24_0 ...
remote: 
remote: Create a pull request for 'sdk_24_0' on GitHub by visiting:        
remote:      https://github.com/looker-open-source/sdk-codegen/pull/new/sdk_24_0        
remote: 
remote: GitHub found 11 vulnerabilities on looker-open-source/sdk-codegen's default branch (5 high, 6 moderate). To find out more, visit:        
remote:      https://github.com/looker-open-source/sdk-codegen/security/dependabot         
remote: 
To github.com:looker-open-source/sdk-codegen.git
 * [new branch]        sdk_24_0 -> sdk_24_0
```

**Comment by drstrangelooker** [comment]
Trying 
```
$ npm run lint:es

> sdk-codegen-all@23.21.0 lint:es
> eslint 'packages/**/*.ts{,x}' --cache
```

It is running for a very long time, might be hanging.

File: python/looker_sdk/sdk/api40/methods.py [hunk 1]
```diff
@@ -112,20 +112,9 @@ def search_alerts(
...
-                    "condition_met": condition_met,
-                    "last_run_start": last_run_start,
-                    "last_run_end": last_run_end,
-                    "all_owners": all_owners,
-                },
-                transport_options=transport_options,
-            ),
+                query_params={"limit": limit, "offset": offset, "group_by": group_by, "fields": fields, "disabled": disabled, "frequency": frequency, "condition_met": condition_met, "last_run_start": last_run_start, "last_run_end": last_run_end, "all_owners": all_owners},
+                transport_options=transport_options
+            )
```
Comment (drstrangelooker) [inline python/looker_sdk/sdk/api40/methods.py:117]: @jeremytchang @jkaster Here is another typical issue with newlines and commas after the last element in a list.

File: kotlin/src/main/com/looker/sdk/4.0/methods.kt [hunk 2]
```diff
@@ -254,17 +259,14 @@ class LookerSDK(authSession: AuthSession) : APIMethods(authSession) {
...
-            mapOf(
-                "limit" to limit,
-                "offset" to offset,
-            ),
-        )
+        offset: Long? = null
+    ) : SDKResponse {
+        return this.get<Array<AlertNotifications>>("/alert_notifications", 
+            mapOf("limit" to limit,
+                 "offset" to offset))
```
[the remark this answers is not in the document]
Comment (drstrangelooker) [reply kotlin/src/main/com/looker/sdk/4.0/methods.kt]: `./gradlew spotlessApply` fixed this

File: python/looker_sdk/sdk/api40/methods.py (hunk 1 above)
Comment (drstrangelooker) [reply python/looker_sdk/sdk/api40/methods.py]: Resolved with `pipenv run black api40/*.py`

**Review by jeremytchang — APPROVED** [review]
LGTM

**Pull request merged**